### ISSUE TITLE
Refactor Annotation_hasFlags

### ISSUE THREAD

**Issue by yurydelendik** [mozilla/pdf.js#7280]
The logic at https://github.com/mozilla/pdf.js/blob/master/src/core/annotation.js#L244 may not work all the time:

```
      if (this.flags) {
        return (this.flags & flag) > 0;
      }
      return false;
```

Let's make check of this.flags !== undefined explicit or make the `this.flags` property set all the time (the latter is better). Also if `flags` using MSB the result will be negative, so it will be better to use `!== 0` or `!!`.

**Comment by timvandermeij** [comment]
Interesting. Which static analysis tool was used to find this?

**Comment by Snuffleupagus** [comment]
> [...] or make the this.flags property set all the time

That should already be happening, given that when a new `Annotation` is created the `setFlags` method is always called, see https://github.com/mozilla/pdf.js/blob/master/src/core/annotation.js#L175.
Looking at the `setFlags` method, see https://github.com/mozilla/pdf.js/blob/master/src/core/annotation.js#L226, we're already checking that it's an integer (and assigning it a default value otherwise). Perhaps we should also ensure that it's positive?

So I suppose that the question is if the `if (this.flags) { ... }` checks in various parts of the code is really necessary!?

**Comment by yurydelendik** [comment]
> That should already be happening, given that when a new Annotation is created the setFlags method is always called

Oh, I see -- missed that. Yeah, it will be nice to have it integer and not perform non-zero check.

> So I suppose that the question is if the if (this.flags) { ... } checks in various parts of the code is really necessary!?

No, they don't. We need to be sure about type of an object field most of the time.

**Comment by timvandermeij** [comment]
Fixed by #7352.
